### DISCUSSION TITLE
Spiral MATE edition - Issues with Brisk Menu

### DISCUSSION THREAD

**Discussion by bin-linux** [SpiralLinux/SpiralLinux-project#102] [Q&A]
Spiral MATE installation in VM
There seems to be a problem with the Brisk Menu and launching apps requiring authentication.
For example I installed Bleachbit and the entry shows as 'pkexec bleachbit'
This runs fine from terminal and from all 3 of the other standard menus
Same applies to the LightDM config tool
It seems at first glance that Brisk Menu doesn't know how to deal with pkexec.
I use the Compact Menu normally so just reporting this out of interest.

**Comment by geckolinux** [comment]
Hi there, sorry for the slow reply. Hmm that's weird, I was sure I had tested that and had it working. I can confirm the problem with Brisk Menu and the Advanced (Mint) Menu, but only with the SpiralLinux additions such as Gnome Software (root) and Snapper GUI (root). The LightDM customizer and Gparted do work via `pkexec`.

**Reply by geckolinux** [reply]
Also **Bleachbit (as root)** from the Debian repos doesn't work with Brisk Menu or Advanced (Mint) Menu. It appears that the problem is only with applications that have 2 menu entries, one for running as a regular user and another entry to run it as root with `pkexec` authentication. It looks like we'll have to change the `.desktop` files in the Mate edition for the SpiralLinux-specific menu entries to run as root with this hack from https://askubuntu.com/a/1388042 :
```
sh -c "pkexec env DISPLAY=$DISPLAY XAUTHORITY=$XAUTHORITY program-name"

**Reply by bin-linux** [reply]
Interesting - thanks for the info.
Not sure if this helps or not. Set up VM with Debian Testing from netinst iso. Just MATE desktop bare install.
Added Bleachbit - ran Bleachbit (Root) OK from Classic and Compact menus. Added Brisk Menu Applet - it also ran OK - just runs `pkexec bleachbit` as before.
Strange.

**Reply by geckolinux** [reply]
Thanks for testing that. Is your SpiralLinux installation also switched to Debian Testing? Otherwise it might have been fixed in Testing but not in Stable. Also if you still have your VM around could you please confirm which packages were preinstalled that contain `*polkit*` or `*policykit*` in the name?

**Comment by bin-linux** [comment]
The Debian Testing install shows the following in the logs from the online installation process
libpolkit-agent-1-0:amd64 (0.105-33, automatic)
libpolkit-gobject-1-0:amd64 (0.105-33, automatic)
mate-polkit-common:amd64 (1.26.0-1, automatic)
mate-polkit:amd64 (1.26.0-1, automatic)
mate-polkit:amd64 (1.26.0-1, automatic)
policykit-1:amd64 (0.105-33, automatic)
polkitd:amd64 (0.105-33, automatic)

After upgrading Spiral MATE to testing I found that there was some odd behaviour in as much as Firefox was just not rendering some sites properly.  I wiped that to try stock Debian Testing instead to identify whether the OS upgrade was the issue or something else.

**Reply by geckolinux** [reply]
Thanks for confirming that. So there's no `policykit-1-gnome` . It could be that the behaviour will change by making the SpiralLinux `*polkit*` / `*policykit*` package set the same as what the Debian Testing installer uses.

> Firefox was just not rendering some sites properly

That's very likely to just be a Firefox thing. Firefox is pretty self-contained except for some system graphics libraries it uses for hardware rendering. But there is a version bump for `firefox-esr` in Testing, so it very well could have bugs.

**Comment by geckolinux** [comment]
Workaround included for the default menu items in SpiralLinux 11.220925 release.
